src/darker/files.py

```diff
@@ -25,3 +41,49 @@ def find_pyproject_toml(path_search_start: Tuple[str, ...]) -> Optional[str]:
         # We do not have access to the user-level config directory, so ignore it.
         err(f"Ignoring user configuration directory due to {e!r}")
         return None
+
+
+def filter_python_files(
+    paths: Collection[Path],  # pylint: disable=unsubscriptable-object
+    root: Path,
+    formatter: BaseFormatter,
+) -> set[Path]:
+    """Get Python files and explicitly listed files not excluded by Black's config.
+
+    :param paths: Relative file/directory paths from CWD to Python sources
+    :param root: A common root directory for all ``paths``
+    :param formatter: The code re-formatter which provides the configuration containing
+                      the exclude options
+    :return: Paths of files which should be reformatted according to
+             ``black_config``, relative to ``root``.
+
+    """
+    sig = inspect.signature(gen_python_files)
+    # those two exist and are required in black>=21.7b1.dev9
+    kwargs = {"verbose": False, "quiet": False} if "verbose" in sig.parameters else {}
+    # `gitignore=` was replaced with `gitignore_dict=` in black==22.10.1.dev19+gffaaf48
+    for param in sig.parameters:
+        if param == "gitignore":
+            kwargs[param] = None  # type: ignore[assignment]
+        elif param == "gitignore_dict":
+            kwargs[param] = {}  # type: ignore[assignment]
+    absolute_paths = {p.resolve() for p in paths}
+    directories = {p for p in absolute_paths if p.is_dir()}
+    files = {p for p in absolute_paths if p not in directories}
+    files_from_directories = set(
+        gen_python_files(
+            directories,
+            root,
+            include=DEFAULT_INCLUDE_RE,
+            exclude=formatter.get_exclude(DEFAULT_EXCLUDE_RE),
+            extend_exclude=formatter.get_extend_exclude(),
+            force_exclude=formatter.get_force_exclude(),
+            report=Report(),
+            **kwargs,  # type: ignore[arg-type]
+        )
+    )
+    return {p.resolve().relative_to(root) for p in files_from_directories | files}
```

Comment:
More readable could be
return {p.resolve().relative_to(root) for p in  files.union(files_from_directories)}